src/soilId/soilIdSlice.ts

```diff
@@ -103,35 +98,45 @@ const soilIdSlice = createSlice({
     ) => {
       state.status = action.payload;
     },
+    claimKey: (state, action: PayloadAction<SoilIdKey>) => {
+      /* When a cache key is claimed, we increment a counter for that key. */
+      const key = action.payload;
+      state.usages[key] = (state.usages[key] ?? 0) + 1;
+    },
+    releaseKey: (state, action: PayloadAction<SoilIdKey>) => {
+      /*
+       * When a cache key is released, we check to see if it is no longer in use.
+       * If no more usages exist, we remove it from the matches cache.
+       */
+      const key = action.payload;
+      if (key in state.usages) {
+        const count = Math.max(0, state.usages[key] - 1);
+        if (count === 0) {
+          delete state.matches[key];
+        }
+        state.usages[key] = count;
+      }
+    },
```

Comment:
questions: 

- since i am a little nervous about the complexity here without having any tests for it, could we consider never flushing the cache for now? it will get flushed whether we like it or not every time the app closes anyway
- from what i can tell, this eviction strategy means that any time a user leaves a site's dashboard and returns to it the app will need to fetch the matches from the backend again, is that right?

Reply:
1) I am open to a team discussion. Will attempt to add unit tests since this repo does have unit testing already.
2) This is correct.

Reply:
Per discussion with the team, I am now removing this functionality. I have also added some basic unit tests.